src/mount-support.c

```diff
@@ -293,3 +293,72 @@ void setup_slave_mount_namespace()
 		die("can not make make / rslave");
 	}
 }
+
+#define BIND_MAX_LINE_LENGTH	512	// arbitrary
+
+void setup_bind_mounts(const char *appname)
+{
+	debug("setup_bind_mounts %s", appname);
+	FILE *f = NULL;
+	const char *bind_profile_dir = "/var/lib/snapd/bind/profiles/";
+
+	// Note that secure_gettenv will always return NULL when suid, so
+	// SNAPPY_LAUNCHER_BIND_MOUNT_PROFILE_DIR can't be (ab)used in
+	// that case.
+	if (secure_getenv("SNAPPY_LAUNCHER_BIND_MOUNT_PROFILE_DIR") != NULL)
+		bind_profile_dir =
+		    secure_getenv("SNAPPY_LAUNCHER_BIND_MOUNT_PROFILE_DIR");
+
+	char profile_path[512];	// arbitrary path name limit
+	int snprintf_rc =
+	    snprintf(profile_path, sizeof(profile_path), "%s/%s.bind",
+		     bind_profile_dir, appname);
+	if (snprintf_rc < 0 || snprintf_rc >= 512) {
+		errno = 0;
+		die("snprintf returned unexpected value");
+	}
+
+	f = fopen(profile_path, "r");
```

Comment:
Instead of parsing this manually I'd recommend using a glibc function like `getmntent`. This way we can have a very familiar syntax and no parser to write.